include/nfc/ndef/nfc_ndef.rst

```diff
@@ -0,0 +1,176 @@
+.. _nfc_ndef:
+
+Custom messages
+###############
+
+Use the :ref:`nfc_ndef_msg` and :ref:`nfc_ndef_record` modules to create standardized messages and records that are not covered by other modules, or to create your own custom messages.
+
+To generate a custom NDEF message, start by creating the records that make up the message.
+Next, generate an empty message of the required length and add the records to the message.
+Finally, encode and create the message.
+
+By default, NDEF messages are encoded for the Type 2 Tag platform.
+See :ref:`nfc_ndef_format_dox` for a description of the format.
+You can choose to encode the message for the Type 4 Tag platform by setting :option:`CONFIG_NFC_NDEF_MSG_WITH_NLEN` to ``y``.
+In this case, an additional field is added in front of an NDEF message.
+
+You can also encapsulate a message as payload for a record.
+For example, a Handover Request message contains several Alternative Carrier records.
+These records themselves are full messages that contain information about alternative carriers that can be used for further communication.
+
+.. _nfc_ndef_record_gen:
+
+Creating a record descriptor
+============================
+
+You can provide the payload of the record in binary format or in a different format that is then converted into binary format.
+The content of the record is defined in a record descriptor.
+
+Use the :c:macro:`NFC_NDEF_RECORD_BIN_DATA_DEF` macro for binary data or the :c:macro:`NFC_NDEF_GENERIC_RECORD_DESC_DEF` macro for other data to create the record descriptor.
+You must provide header information for the record (for example, ID and type) and the payload data to the macro.
+For generic data that is not in binary format, you must also provide a payload constructor.
+This constructor converts the payload data to the needed format.
+For binary data, the module provides this constructor.
+
+The following code example shows how to generate a record descriptor:
+
+.. code-block:: c
+
+   ret_code_t error_code;
```

Comment:
I have not updated this code block and the one below - please check.

Reply:
a few comments for this one to make it more Zephyr like